bigquery/storage/managedwriter/adapt/protoconversion.go

```diff
@@ -348,12 +347,7 @@ func tableFieldSchemaToFieldDescriptorProto(field *storagepb.TableFieldSchema, i
 
 // nameRequiresAnnotation determines whether a field name requires unicode-annotation.
 func nameRequiresAnnotation(in string) bool {
-	for i := 0; i < len(in); i++ {
-		if in[i] > unicode.MaxASCII {
-			return true
-		}
-	}
-	return false
+	return !protoreflect.Name(in).IsValid()
```

Comment:
cool way to do it, can you please add some tests for cases like: col-1, 00col, [col] with special ascii characters.

Reply:
done